web/src/main/java/org/mskcc/cbio/portal/web/ProxyController.java

```diff
@@ -159,4 +167,85 @@ private String respProxy(String url, HttpMethod method, Object body, HttpServlet
 
     return responseEntity.getBody();
   }
+  
+	@RequestMapping(value = "/virtual-cohort", method = RequestMethod.POST)
```

Comment:
I would suggest that once this pull request has been accepted that we create an issue to merge the bookmarking and virtual cohort methods in this ProxyController class.  I think the virtual cohort methods are generic enough that they could also handle the bookmarking as well.  At least this should be considered further, but probably not done now.  At that time the styles of both might be made more similar (here and in the SessionServiceUtil) and the security from #1971 could be added to the virtual cohort.